server/service/src/common_stock.rs

```diff
@@ -0,0 +1,38 @@
+use repository::{
+    EqualFilter, RepositoryError, StockLine, StockLineFilter, StockLineRepository,
+    StorageConnection,
+};
+
+#[derive(Debug, PartialEq)]
+pub enum CommonStockLineError {
+    DatabaseError(RepositoryError),
+    StockLineDoesNotBelongToStore,
+}
+
+pub fn check_stock_line_exists(
+    connection: &StorageConnection,
+    store_id: &str,
+    id: &str,
+) -> Result<StockLine, CommonStockLineError> {
+    use CommonStockLineError::*;
+
+    let stock_line = StockLineRepository::new(connection)
+        .query_by_filter(
+            StockLineFilter::new().id(EqualFilter::equal_to(id)),
+            Some(store_id.to_string()),
+        )?
+        .pop()
+        .ok_or(DatabaseError(RepositoryError::NotFound))?;
+
+    if stock_line.stock_line_row.store_id != store_id {
+        return Err(StockLineDoesNotBelongToStore);
```

Comment:
Ohh man, the store_id on StockLineRepository is pretty deceiving, can you add a comment please that store_id in the query applies to item not to the stock line, thanks.

Reply:
`apply_equal_filter!(query, store_id, stock_line_dsl::store_id)` ?

item doesn't have a store_id, or am I missing something?

Reply:
The store_id parameter that is being set in query_by_filter for StockLineRepository is only use in apply_item_filter, and that in turn is used for item visibility. We should probably change that logic eventually, actually can you change Some(store_id.to_sting()), in None in this case